kalite/distributed/features/steps/superuser_create.py

```diff
@@ -0,0 +1,145 @@
+def step_impl(context):
+    fill_username(context, "")
+
+@when('I click the create button')
+def impl(context):
+    create_button = find_css_class_with_wait(context, "create-btn")
+    create_button.click()
+
+@then('the username border will turn red')
+def impl(context):
+    is_border_red(context, "super_username")
+
+@given("I enter a username longer than 30 letters")
+def step_impl(context):
+    fill_username(context, "123456789123456789123456789123456789")
+
+@when('I click the create button')
+def impl(context):
+    create_button = find_css_class_with_wait(context, "create-btn")
+    create_button.click()
+
+@then("the modal won't dismiss")
+def impl(context):
+    assert find_id_with_wait(context, "superusercreate-container")
+
+@given("the password is empty")
+def step_impl(context):
+    fill_password(context, "")
+
+@when('I click the create button')
+def impl(context):
+    create_button = find_css_class_with_wait(context, "create-btn")
+    create_button.click()
+
+@then('the password border will turn red')
+def impl(context):
+    is_border_red(context, "super_password")
+
+@given("I enter a password longer than 30 letters")
+def step_impl(context):
+    fill_password(context, "123456789123456789123456789123456789")
+
+@when('I click the create button')
+def impl(context):
+    create_button = find_css_class_with_wait(context, "create-btn")
+    create_button.click()
+
+@then("the modal won't dismiss")
+def impl(context):
+    assert find_id_with_wait(context, "superusercreate-container")
+
+@given("I enter my email address incorrectly")
+def step_impl(context):
+    fill_email(context, "incorrect")
+
+@when('I click the create button')
+def impl(context):
+    create_button = find_css_class_with_wait(context, "create-btn")
+    create_button.click()
+
+@then("the email border will turn red")
+def impl(context):
+    is_border_red(context, "super_email")
+
+@given("I enter a email address longer than 30 letters")
+def step_impl(context):
+    fill_email(context, "123456789123456789123456789@toolong.com")
+
+@when('I click the create button')
+def impl(context):
+    create_button = find_css_class_with_wait(context, "create-btn")
+    create_button.click()
+
+@then("the modal won't dismiss")
+def impl(context):
+    assert find_id_with_wait(context, "superusercreate-container")
+
+@given("I enter everything correctly")
+def step_impl(context):
+    fill_email(context, "correct@email.com")
+    fill_username(context, "correct_name")
+    fill_password(context, "correct_password")
+
+@when('I click the create button')
+def impl(context):
+    create_button = find_css_class_with_wait(context, "create-btn")
+    create_button.click()
+
+@then("the modal will dismiss")
+def impl(context):
+    assert not find_id_with_wait(context, "superusercreate-container")
+    assert User.objects.exists()
+
+
+
+def fill_field(context, text, field_id):
+    field = find_id_with_wait(context, field_id)
+    field.send_keys(text)
```

Comment:
`find_id_with_wait` returns `None` if the element isn't found, right? I think it makes sense to assert that `field is not None` here (with an appropriate message), otherwise an error will be raised by line 131 that will be somewhat less than helpful.